Scripts/Runtime/Util/Jobs/ParallelCollectionUtil.cs

```diff
@@ -0,0 +1,103 @@
+using System;
+using System.Reflection;
+using Unity.Jobs.LowLevel.Unsafe;
+using UnityEngine;
+
+namespace Anvil.Unity.DOTS.Util
+{
+    /// <summary>
+    /// Utility methods for working with Native Collections in a parallel manner.
+    /// </summary>
+    public static class ParallelCollectionUtil
+    {
+        /// <remarks>
+        /// This is static readonly because Burst will only read from a static variable if it is readonly.
+        /// We are unable to set it directly to <see cref="JobsUtility.JobWorkerMaximumCount"/> because that function
+        /// is an external function and external functions are not allowed to be called from a static constructor which
+        /// is where static readonly variables get assigned.
+        /// We are also unable to lazy instantiate or call <see cref="JobsUtility.JobWorkerMaximumCount"/> within the
+        /// functions in this class that use it because an external function can only be called from the main thread
+        /// and these functions are intended for use inside bursted jobs that run on many different worker threads.
+        ///
+        /// To avoid this, we have a <see cref="RuntimeInitializeOnLoadMethodAttribute"/> on the
+        /// <see cref="ParallelCollectionUtil.Init"/> method which uses reflection to assign the static readonly
+        /// variable.
+        ///
+        /// While a bit hacky, it is ensured to run before any potential jobs execute. The external call is allowed
+        /// and we get the correct value and inject it into the static readonly field. Subsequent bursted jobs are then
+        /// allowed to access it.
+        /// </remarks>
+        private static readonly int JOB_WORKER_MAXIMUM_COUNT = -1;
+
+        [RuntimeInitializeOnLoadMethod(RuntimeInitializeLoadType.SubsystemRegistration)]
+        private static void Init()
+        {
+            Type type = typeof(ParallelCollectionUtil);
+            FieldInfo field = type.GetField(nameof(JOB_WORKER_MAXIMUM_COUNT), BindingFlags.Static | BindingFlags.NonPublic);
+            field.SetValue(null, JobsUtility.JobWorkerMaximumCount);
+        }
+
+        /// <summary>
+        /// Returns an ideal size for the number of buckets a collection should have
+        /// to account for all possible threads that could write to it at once.
+        /// </summary>
+        /// <remarks>
+        /// There is a lot of different terminology for the "buckets".
+        /// <see cref="NativeStream"/> has foreachCount
+        /// <see cref="UnsafeTypedStream"/> has lanes
+        /// etc
+        /// It's the number of separate "buckets" that can be written to in parallel.
+        /// </remarks>
+        /// <returns>The number of buckets a collection should have.</returns>
+        public static int CollectionSizeForMaxThreads()
+        {
+            return JOB_WORKER_MAXIMUM_COUNT + 1;
+        }
```

Comment:
Should just be a getter or maybe a `public static readonly` value?

Reply:
Can't do that because of when I get to assign `JOB_WORKER_MAXIMUM_COUNT` but I did move to a getter/private setter and init in the `Init` function instead.